src/native/fs/fs_napi.cpp

```diff
@@ -156,12 +157,14 @@ struct FSWorker : public Napi::AsyncWorker
     FSWorker(const Napi::CallbackInfo& info)
         : AsyncWorker(info.Env())
         , _deferred(Napi::Promise::Deferred::New(info.Env()))
+        , _args_ref(Napi::Persistent(Napi::Object::New(info.Env())))
         , _tid(0)
         , _uid(ThreadScope::orig_uid)
         , _gid(ThreadScope::orig_gid)
         , _errno(0)
         , _warn_threshold_ms(0)
     {
+        for (int i = 0; i < (int)info.Length(); ++i) _args_ref.Set(i, info[i]);
```

Comment:
Actually - we need to do the same for CryptoWorker !
Perhaps this can explain why we saw bad md5 calculations..?

https://github.com/noobaa/noobaa-core/blob/2b5600631c0c4964e35a966e889c76e19709606e/src/native/tools/crypto_napi.cpp#L22-L25